### PR TITLE
Add unit tests for queries using NONE.

### LINKED ISSUE #1886: Create unit test for NONE predicate

### PR DESCRIPTION
**Pull request by bdash**
Fixes #1886.

/cc @yoshyosh @jpsim

### CODE REVIEW

File: Realm/Tests/QueryTests.m [hunk 1]
```diff
@@ -457,6 +457,9 @@ - (void)testPredicateNotSupported
 
     // ALL
     XCTAssertThrows([realm objects:className where:@"ALL intCol > 5"], @"ALL int > constant");
+
+    // ANY
```
Comment (jpsim) [inline Realm/Tests/QueryTests.m]: NONE

**Comment by jpsim** [comment]
:+1:
